src/Digdir.Domain.Dialogporten.Application/Features/V1/EndUser/Dialogs/Queries/Get/GetDialogQuery.cs

```diff
@@ -157,15 +157,15 @@ public async Task<GetDialogResult> Handle(GetDialogQuery request, CancellationTo
                 }
             }
 
-            // Simple "read" on the main resource will give access to a dialog element, unless an authorization attribute is set,
-            // in which case an "elementread" action is required
             var elements = dto.Elements.Where(dialogElement =>
-                (dialogElement.AuthorizationAttribute is null && action == Constants.ReadAction) ||
-                (dialogElement.AuthorizationAttribute is not null && action == Constants.ElementReadAction));
+                // If no authorization attribute is set, we just required "read" on the main resource
+                (dialogElement.AuthorizationAttribute is null && action == Constants.ReadAction && resource == Constants.MainResource) ||
+                // If an authorization attribute is set, we require either "elementread" (if subresource) or "read" (if external resource)
+                (dialogElement.AuthorizationAttribute == resource && action is Constants.ElementReadAction or Constants.ReadAction));
```

Comment:
Does this logic match the comment? 
Seems the comment means to check for `elementread` if subresource, and `read` if external resource?
🤔

Reply:
At this point, either read or elementread works, but the comment is unclear. Changed this to use the extension methods in `DialogDetailsAuthorizationResult` (which is used in the dialog element endpoint)